samples/sample_encode/src/pipeline_encode.cpp

```diff
@@ -1663,6 +1719,62 @@ mfxStatus CEncodingPipeline::ResetMFXComponents(sInputParams* pParams)
     return MFX_ERR_NONE;
 }
 
+mfxStatus CEncodingPipeline::AllocExtBuffers(sInputParams *pInParams)
+{
+    MSDK_CHECK_POINTER(pInParams, MFX_ERR_NULL_PTR);
+    mfxStatus sts = MFX_ERR_NONE;
+
+    bufSet* tmpForInit   = NULL;
+    mfxExtAVCRoundingOffset* m_AvcRoundingOffset = NULL;
+
+    mfxU16  numOfFields  = pInParams->nPicStruct != MFX_PICSTRUCT_PROGRESSIVE ? 2 : 1;
+    mfxU16  numOfBuffers = pInParams->nGopRefDist * 2 + pInParams->nAsyncDepth + pInParams->nNumRefFrame + 1;
+
+    bool enableRoundingOffset = pInParams->RoundingOffsetFile && pInParams->CodecId == MFX_CODEC_AVC;
+    if (enableRoundingOffset && m_round_in == NULL)
+    {
+        MSDK_FOPEN(m_round_in, pInParams->RoundingOffsetFile, MSDK_CHAR("rb"));
+        if (m_round_in == NULL) {
+            msdk_printf(MSDK_STRING("ERROR: Can't open file %s\n"), pInParams->RoundingOffsetFile);
+            return MFX_ERR_UNSUPPORTED;
+        }
+
+        msdk_printf(MSDK_STRING("Using rounding offset input file: %s\n"), pInParams->RoundingOffsetFile);
+    }
+
+    for (int k = 0; k < numOfBuffers; k++)
```

Comment:
you shouldn't create buffers set if no buffers are actually required

Reply:
OK